librz/util/str.c

```diff
@@ -699,7 +699,6 @@ RZ_API const char *rz_str_nstr(const char *s, const char *find, int slen) {
 }
 
 // Returns a new heap-allocated copy of str.
-// XXX what's the diff with rz_str_dup ?
 RZ_API char *rz_str_new(const char *str) {
```

Comment:
`rz_str_new` takes 1 argument but you just renamed all `rz_str_dup` to that but not changed the arguments.

Reply:
also, the old `rz_str_dup` was also freeing the `ptr` argument, which means that now all things will leak.